packages/mui-base/src/utils/getElementDocument.ts

```diff
@@ -0,0 +1,8 @@
+/**
+ * Get the document associated with an element.
+ * @param element
+ * @returns `Document` associated with the element
+ */
+export function getElementDocument(element: Element | null) {
+  return element?.ownerDocument || document;
```

Comment:
We have an `ownerDocument` on `@mui/utils` which does exactly this

Reply:
Thank you. Fixed!